### PR TITLE
feat: Use https scheme for IANA target URLs

### LINKED ISSUE #417: IANA refs should use https URLs
### Describe the issue

Currently, IANA references created using the BibXML service use http for URLs, while the style guide recommends using https [1]. 

[1] https://www.rfc-editor.org/styleguide/part2/#ref_iana_reg

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by kesara**
Fixes #417

### CODE REVIEW

File: xml2rfc_compat/tests/test_adapters.py [hunk 1]
```diff
@@ -161,3 +161,16 @@ def test_doi_should_replace_target_URL(self):
...
+        bibitem = adapter.resolve()
+        for link in as_list(bibitem.link or []):
+            if content := link.__getattribute__("content"):
+                url_parse = urlparse(content)
+                self.assertEqual(url_parse.scheme, "https")
+        self.assertTrue(any(
+            link
+            for link in as_list(bibitem.link or [])
+            if urlparse(link.content).scheme == "https"
+        ))
```
Comment (microamp) [inline xml2rfc_compat/tests/test_adapters.py:176]: Can this also be
```python
self.assertTrue(
    all(
        urlparse(link.content).scheme == "https"
        for link in as_list(bibitem.link or [])
    )
)
```
?

File: bibxml/xml2rfc_adapters.py [hunk 2]
```diff
@@ -503,6 +503,23 @@ def resolve_docid(self) -> Optional[DocID]:
...
+        refs = self.fetch_refs()
+        if num_refs := len(refs):
+            self.log(f"{num_refs} found")
+            resolved_item = self.build_bibitem_from_refs(refs)
+            link = as_list(resolved_item.link or [])
+            for index, _ in enumerate(link):
+                parsed_link = urlparse(link[index].content)
+                if parsed_link.scheme == "http":
+                    link[index].content = \
+                        parsed_link._replace(scheme="https").geturl()
```
Comment (microamp) [inline bibxml/xml2rfc_adapters.py:516]: I wonder if there's a public method for this (i.e. `_replace`). Please ignore otherwise.

**Review by microamp — APPROVED** [review]
Looks good to me :ship:

File: xml2rfc_compat/tests/test_adapters.py (hunk 1 above)
Comment (kesara) [reply xml2rfc_compat/tests/test_adapters.py]: Suggestion accepted.

File: bibxml/xml2rfc_adapters.py (hunk 2 above)
Comment (kesara) [reply bibxml/xml2rfc_adapters.py:516]: Python documentation is using urllib documentation says no? https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse

**Pull request merged**